src/fragments/lib/storage/ios/download.mdx

```diff
@@ -6,37 +6,52 @@ You can download to in-memory buffer [Data](https://developer.apple.com/document
 
 <BlockSwitcher>
 
-<Block name="Listener (iOS 11+)">
+<Block name="Async/Await">
 
 ```swift
-let storageOperation = Amplify.Storage.downloadData(
-    key: "myKey", 
-    progressListener: { progress in
-        print("Progress: \(progress)")
-    }, resultListener: { (event) in
-        switch event {
-        case let .success(data):
-            print("Completed: \(data)")
-        case let .failure(storageError):
-            print("Failed: \(storageError.errorDescription). \(storageError.recoverySuggestion)")
+do {
+    let downloadTask = try await Amplify.Storage.downloadData(key: "myKey")
+    Task {
+        for await progress in await downloadTask.progress {
+            print("Progress: \(progress)")
+        }
     }
-})
+    let data = try await downloadTask.value
+    print("Completed: \(data)")
+} catch let error as StorageError {
+    print("Failed: \(error.errorDescription). \(error.recoverySuggestion)")
+} catch {
+    print("Failed: \(error)")
+}
 ```
 
 </Block>
 
 <Block name="Combine">
 
 ```swift
-let storageOperation = Amplify.Storage.downloadData(key: "myKey")
-let progressSink = storageOperation.progressPublisher.sink { progress in print("Progress: \(progress)") }
-let resultSink = storageOperation.resultPublisher.sink {
-    if case let .failure(storageError) = $0 {
-        print("Failed: \(storageError.errorDescription). \(storageError.recoverySuggestion)")
-    }
-}
-receiveValue: { data in
-    print("Completed: \(data)")
+do {
```

Comment:
this code could drop the `do/catch` too since that is boilerplate which we do not have to repeat in docs